tools/rln_keystore_generator/rln_keystore_generator.nim

```diff
@@ -50,17 +51,44 @@ when isMainModule:
                        idSecretHash = credential.idSecretHash.inHex(),
                        idCommitment = credential.idCommitment.inHex()
 
-  # 4. write to keystore
-  ## TODO: after hooking up to the OnchainGroupManager, 
-  ## obtain chainId and treeIndex from the contract
+  
+  if not conf.execute:
+    info "not executing, exiting"
+    quit(0)
+
+  # 4. initialize OnchainGroupManager
+  let groupManager = OnchainGroupManager(ethClientUrl: conf.rlnRelayEthClientAddress,
+                                         ethContractAddress: conf.rlnRelayEthContractAddress,
+                                         rlnInstance: rlnInstance,
+                                         keystorePath: none(string),
+                                         keystorePassword: none(string),
+                                         ethPrivateKey: some(conf.rlnRelayEthPrivateKey),
+                                         # saveKeystore = false, since we're managing it
+                                         saveKeystore: false)
+  try:
+    waitFor groupManager.init()
+  except CatchableError:
+    error "failure while initializing OnchainGroupManager", error=getCurrentExceptionMsg()
+    quit(1)
+
+  # 5. register on-chain
+  try:
+    waitFor groupManager.register(credential)
```

Comment:
Solving this is not related to this PR but if we have a tool to register a membership in nwaku perhaps we should fix this https://github.com/status-im/nim-web3/issues/79 ? May take some time though.

I don't think you are able to broadcast a tx without a chaindId so this `groupManager.register` will most likely fail unless your Ethereum node uses `rpc.allow-unprotected-txs`, and afaik thats disabled in most of the cases. So someone using this tool will most likely run into this.

Reply:
very nice catch! will try to make some time to fix that!